crates/noirc_frontend/src/ast/structure.rs

```diff
@@ -33,6 +33,23 @@ pub struct NoirImpl {
     pub methods: Vec<NoirFunction>,
 }
 
+/// Ast node for type aliases
+#[derive(Clone, Debug)]
+pub struct NoirTyAlias {
+    pub name: Ident,
+    pub ty: UnresolvedType,
+    pub span: Span,
+    // TODO: should probabaly allow generics
+    // eg. type foo = Vec<T>;
+    // pub generics: UnresolvedGenerics,
```

Comment:
We should keep this in mind when the PR is nearing completion, type aliases should support generics as well. I can help with this if needed.